gateway/src/main/java/org/georchestra/gateway/events/EventsConfiguration.java

```diff
@@ -0,0 +1,36 @@
+package org.georchestra.gateway.events;
+
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.listener.MessageListenerContainer;
+import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.amqp.core.Queue;
+import org.springframework.cloud.gateway.config.GatewayAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
+import org.springframework.context.annotation.Profile;
+
+@Profile("!test && !it")
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureAfter(GatewayAutoConfiguration.class)
+@ImportResource({ "classpath:rabbit-listener-context.xml", "classpath:rabbit-sender-context.xml" })
+public class EventsConfiguration {
```

Comment:
Rename as something more concrete and that has AutoConfiguration suffix to follow spring boot converntions, e.g. OAuth2EventsAutoConfiguration.

You should add some conditional here. We still need to run the gateway in environments where rabbit is not needed/present. As is stands, that would lead to a startup failure.

What's the mechanism to enable/disable the account notifications at all? maybe follow the same pattern or use a @ConditionalOnProfile

Reply:
@fvanderbiest  I think this point should be discussed in group, should gateway be dependent over rabbitmq or not ? 
In other words, should gateway continue its job even if there is no rabbitmq ?

Reply:
To me it is pretty clear that it shouldn't be depending on this functionality because we use gateway or intend to use it in other (non georchestra specific) environments, such as geoserver cloud.

Reply:
@marwanehcine `@ConditionalOnExpression("${georchestra.gateway.security.createNonExistingUsersInLDAP:true}")` is good enough for me.